kimchi/src/circuits/expr.rs

```diff
@@ -3149,3 +3149,246 @@ pub mod test {
         assert_eq!(test_4::<Fp, Fp>(Fp::from(5u64)), Fp::from(160u64));
     }
 }
+
+mod quadricization {
+    use super::{ColTrait, Expr, Op2, Variable};
+    use crate::circuits::gate::CurrOrNext;
+    use ark_ff::{One, Zero};
+    use std::{
+        collections::{BTreeMap, HashMap},
+        fmt::Debug,
+        hash::Hash,
+        ops::Neg,
+    };
+
+    ///extension of columns with necessary extra columns
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+    pub enum ColumnExtended<S: ColTrait + Eq + Hash> {
+        Inner(S),
+        ///extra column used to reduce degree of expressions
+        Extended(usize),
```

Comment:
Would it make sense to document that `usize` means scaling coefficient (if I understand correctly?).